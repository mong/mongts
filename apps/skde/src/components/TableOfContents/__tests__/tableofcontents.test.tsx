/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render, fireEvent } from "@testing-library/react";
import { TableOfContents } from "..";
import React from "react";
import { test, expect } from "vitest";

const resultatbox1 = {
  overskrift: "Kirurgiske inngrep for endometriose",
  type: "resultatboks",
  tekst: "test",
};

const resultatbox2 = {
  overskrift: "Kirurgiske inngrep for noe annet",
  type: "resultatboks",
  tekst: "test",
};

const obj = [
  {
    overskrift: "Test en",
    innhold: [
      { type: "tekst", tekst: "Teksten min\n# Overskrift h1\nLitt mer tekst" },
    ],
  },
  { overskrift: "Test to", innhold: [resultatbox1, resultatbox2] },
  { overskrift: "Test tre", innhold: [{ type: "tekst", tekst: "Tekst" }] },
  { overskrift: "Test fire", innhold: [{ type: "tekst", tekst: "Tekst" }] },
];

const tocData = obj
  .filter((chapter) => chapter.overskrift)
  .map((chapter) => {
    const level = 1;
    const elemID = chapter.overskrift;
    const children = chapter.innhold
      ? chapter.innhold
          .filter((subChapter) => subChapter.type === "resultatboks")
          .map((subChapter) => ({
            level: 2,
            elemID: subChapter["overskrift"],
            children: [],
          }))
      : [];
    return { level, elemID, children };
  });

test("English menu", async () => {
  const { container, getAllByTestId } = render(
    <TableOfContents lang="en" tocData={tocData} />,
  );
  const firstSnap = container;
  expect(container).toMatchSnapshot();
  // Expand
  fireEvent.click(getAllByTestId("toc_a_#test-to")[0]);
  expect(container).toMatchSnapshot();
  // Reduce
  fireEvent.click(getAllByTestId("toc_a_#test-to")[0]);
  expect(container).toEqual(firstSnap);
});

test("Norwegian menu", async () => {
  const { container } = render(<TableOfContents lang="no" tocData={tocData} />);
  expect(container).toMatchSnapshot();
});

test("Another tocData", async () => {
  const tocData = [
    {
      level: 1,
      elemID: "test",
      children: [
        {
          level: 2,
          elemID: "test-child",
          children: [],
        },
      ],
    },
  ];
  const { container, getByTestId } = render(
    <TableOfContents tocData={tocData} />,
  );
  expect(container).toMatchSnapshot();
  const accordionSummary = getByTestId("toc-header");
  const accordionDetails = getByTestId("toc-content");

  expect(accordionSummary).toBeTruthy();
  expect(accordionDetails).toBeTruthy();
});
