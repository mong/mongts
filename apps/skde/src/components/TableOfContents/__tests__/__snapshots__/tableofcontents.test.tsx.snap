// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Another tocData 1`] = `
<div>
  <div
    class="_toc_large_3064e4"
  >
    <div
      class="_tocWrapper_3064e4"
    >
      <nav
        class="_toc_3064e4"
      >
        <ol
          class="_tocList_3064e4"
        >
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test"
              href="#test"
            >
              test
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="_toc_small_wrapper_3064e4"
  >
    <div
      class="_toc_small_content_3064e4"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gso84h-MuiPaper-root-MuiAccordion-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h3
          class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
        >
          <div
            aria-controls="toc-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-z56f07-MuiButtonBase-root-MuiAccordionSummary-root"
            data-testid="toc-header"
            id="toc-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-7iurs7-MuiAccordionSummary-content"
            >
              Innholdsoversikt
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                color="#034584"
                fill="currentColor"
                font-size="large"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(3, 69, 132);"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </h3>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="toc-header"
                class="MuiAccordion-region"
                id="toc-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  data-testid="toc-content"
                >
                  <nav
                    class="_toc_3064e4"
                  >
                    <ol
                      class="_tocList_3064e4"
                    >
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test"
                          href="#test"
                        >
                          test
                        </a>
                      </li>
                    </ol>
                  </nav>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`English menu 1`] = `
<div>
  <div
    class="_toc_large_3064e4"
  >
    <div
      class="_tocWrapper_3064e4"
    >
      <nav
        class="_toc_3064e4"
      >
        <ol
          class="_tocList_3064e4"
        >
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-en"
              href="#test-en"
            >
              Test en
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-to"
              href="#test-to"
            >
              Test to
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-tre"
              href="#test-tre"
            >
              Test tre
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-fire"
              href="#test-fire"
            >
              Test fire
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="_toc_small_wrapper_3064e4"
  >
    <div
      class="_toc_small_content_3064e4"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gso84h-MuiPaper-root-MuiAccordion-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h3
          class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
        >
          <div
            aria-controls="toc-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-z56f07-MuiButtonBase-root-MuiAccordionSummary-root"
            data-testid="toc-header"
            id="toc-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-7iurs7-MuiAccordionSummary-content"
            >
              Table of contents
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                color="#034584"
                fill="currentColor"
                font-size="large"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(3, 69, 132);"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </h3>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="toc-header"
                class="MuiAccordion-region"
                id="toc-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  data-testid="toc-content"
                >
                  <nav
                    class="_toc_3064e4"
                  >
                    <ol
                      class="_tocList_3064e4"
                    >
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-en"
                          href="#test-en"
                        >
                          Test en
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-to"
                          href="#test-to"
                        >
                          Test to
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-tre"
                          href="#test-tre"
                        >
                          Test tre
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-fire"
                          href="#test-fire"
                        >
                          Test fire
                        </a>
                      </li>
                    </ol>
                  </nav>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`English menu 2`] = `
<div>
  <div
    class="_toc_large_3064e4"
  >
    <div
      class="_tocWrapper_3064e4"
    >
      <nav
        class="_toc_3064e4"
      >
        <ol
          class="_tocList_3064e4"
        >
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-en"
              href="#test-en"
            >
              Test en
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-to"
              href="#test-to"
            >
              Test to
            </a>
            <ol
              class="_tocList_3064e4"
            >
              <li
                class="_tocListItem_3a2e40 false"
                data-testid="tocItem"
              >
                <a
                  data-testid="toc_a_#test-to_kirurgiske-inngrep-for-endometriose"
                  href="#test-to_kirurgiske-inngrep-for-endometriose"
                >
                  Kirurgiske inngrep for endometriose
                </a>
              </li>
              <li
                class="_tocListItem_3a2e40 false"
                data-testid="tocItem"
              >
                <a
                  data-testid="toc_a_#test-to_kirurgiske-inngrep-for-noe-annet"
                  href="#test-to_kirurgiske-inngrep-for-noe-annet"
                >
                  Kirurgiske inngrep for noe annet
                </a>
              </li>
            </ol>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-tre"
              href="#test-tre"
            >
              Test tre
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-fire"
              href="#test-fire"
            >
              Test fire
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="_toc_small_wrapper_3064e4"
  >
    <div
      class="_toc_small_content_3064e4"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gso84h-MuiPaper-root-MuiAccordion-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h3
          class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
        >
          <div
            aria-controls="toc-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-z56f07-MuiButtonBase-root-MuiAccordionSummary-root"
            data-testid="toc-header"
            id="toc-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-7iurs7-MuiAccordionSummary-content"
            >
              Table of contents
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                color="#034584"
                fill="currentColor"
                font-size="large"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(3, 69, 132);"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </h3>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="toc-header"
                class="MuiAccordion-region"
                id="toc-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  data-testid="toc-content"
                >
                  <nav
                    class="_toc_3064e4"
                  >
                    <ol
                      class="_tocList_3064e4"
                    >
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-en"
                          href="#test-en"
                        >
                          Test en
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-to"
                          href="#test-to"
                        >
                          Test to
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-tre"
                          href="#test-tre"
                        >
                          Test tre
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-fire"
                          href="#test-fire"
                        >
                          Test fire
                        </a>
                      </li>
                    </ol>
                  </nav>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Norwegian menu 1`] = `
<div>
  <div
    class="_toc_large_3064e4"
  >
    <div
      class="_tocWrapper_3064e4"
    >
      <nav
        class="_toc_3064e4"
      >
        <ol
          class="_tocList_3064e4"
        >
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-en"
              href="#test-en"
            >
              Test en
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-to"
              href="#test-to"
            >
              Test to
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-tre"
              href="#test-tre"
            >
              Test tre
            </a>
          </li>
          <li
            class="_tocListItem_3a2e40 false"
            data-testid="tocItem"
          >
            <a
              data-testid="toc_a_#test-fire"
              href="#test-fire"
            >
              Test fire
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="_toc_small_wrapper_3064e4"
  >
    <div
      class="_toc_small_content_3064e4"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1gso84h-MuiPaper-root-MuiAccordion-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h3
          class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
        >
          <div
            aria-controls="toc-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-z56f07-MuiButtonBase-root-MuiAccordionSummary-root"
            data-testid="toc-header"
            id="toc-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-7iurs7-MuiAccordionSummary-content"
            >
              Innholdsoversikt
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                color="#034584"
                fill="currentColor"
                font-size="large"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(3, 69, 132);"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </h3>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="toc-header"
                class="MuiAccordion-region"
                id="toc-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  data-testid="toc-content"
                >
                  <nav
                    class="_toc_3064e4"
                  >
                    <ol
                      class="_tocList_3064e4"
                    >
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-en"
                          href="#test-en"
                        >
                          Test en
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-to"
                          href="#test-to"
                        >
                          Test to
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-tre"
                          href="#test-tre"
                        >
                          Test tre
                        </a>
                      </li>
                      <li
                        class="_tocListItem_3a2e40 false"
                        data-testid="tocItem"
                      >
                        <a
                          data-testid="toc_a_#test-fire"
                          href="#test-fire"
                        >
                          Test fire
                        </a>
                      </li>
                    </ol>
                  </nav>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
